lib/routes/routeBackbeat.js

```diff
@@ -578,62 +616,99 @@ function putMetadata(request, response, bucketInfo, objMd, log, callback) {
+            // might point to a non current version. To prevent this, we need to set repairMaster to
+            // true which will update the master to the latest version available at the time of the put.
+            // The update is only done when putting a new version as updating versions that already exist
+            // shouldn't affect the master.
+            if (!objMd) {
+                options.repairMaster = true;
+            }
         }
 
         // If the new null keys logic (S3C-7352) is not supported (compatibility mode), 'isNull' remains undefined.
         if (!nullVersionCompatMode) {
             options.isNull = isNull;
         }
 
-        log.trace('putting object version', {
-            objectKey: request.objectKey, omVal, options });
-        return metadata.putObjectMD(bucketName, objectKey, omVal, options, log,
-            (err, md) => {
-                if (err) {
-                    log.error('error putting object metadata', {
-                        error: err,
-                        method: 'putMetadata',
-                    });
-                    return callback(err);
+        return async.series([
+            // Zenko's CRR delegates replacing the account
+            // information to the destination's Cloudserver, as
+            // Vault admin APIs are not exposed externally.
+            next => {
+                // Internal users of this API (other features in Zenko) will
+                // not provide the accountId in the request, as they only update
+                // the metadata of existing objects, so there is no need to
+                // replace the account information.
+                if (!request.query?.accountId) {
+                    return next();
                 }
-                pushReplicationMetric(objMd, omVal, bucketName, objectKey, log);
-                if (objMd &&
-                    headers['x-scal-replication-content'] !== 'METADATA' &&
-                    versionId &&
-                    // The new data location is set to null when archiving to a Cold site.
-                    // In that case "removing old data location key" is handled by the lifecycle
-                    // transition processor.
-                    omVal.location && Array.isArray(omVal.location) &&
-                    locationKeysHaveChanged(objMd.location, omVal.location)) {
-                    log.info('removing old data locations', {
-                        method: 'putMetadata',
-                        bucketName,
-                        objectKey,
-                    });
-                    async.eachLimit(objMd.location, 5,
-                    (loc, next) => dataWrapper.data.delete(loc, log, err => {
+                return getCanonicalIdsByAccountId(request.query.accountId, log, (err, res) => {
+                    if (err) {
+                        return next(err);
+                    }
+                    if (res) {
```

Comment:
Isn't `res` always set as per the checks in `getCanonicalIdsByAccountId`? THen this if is not needed.

By the way, shouldn't `getCanonicalIdsByAccountId` be a method in arsenal, more than a function in the backbeat routes? In case we need this primitive from other places later...